src/Appwrite/Platform/Modules/Sites/Http/Variables/Update.php

```diff
@@ -74,9 +76,14 @@ public function action(string $siteId, string $variableId, string $key, ?string
             throw new Exception(Exception::VARIABLE_NOT_FOUND);
         }
 
+        if ($variable->getAttribute('secret') && !$secret) {
```

Comment:
let's do === true and === false, to make sure we don't get random nulls or zeros triggering weird behaviour